es6-lib/soql/soql.js

```diff
@@ -14,9 +14,27 @@ class SoQL {
    * TODO: this is necessary, but is this sufficient?
    */
   _launderName(name) {
+    name = name.trim();
+    var start = name[0];
+    var isNumber = !_.isNaN(parseInt(start));
+    var isUpper = start.toUpperCase() === start;
+
+
+    // Suppose we have columns in one layer
+    // called `NAME` and `name`, snakecase itself will convert
+    // these to `name` and `name`. So we need to convert them
```

Comment:
i'm punting on that for now - i'd want to think a little bit more on where that logic would go. this is enough to unblock the people who have duplicate columns due to case